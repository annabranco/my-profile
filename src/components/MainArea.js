import React from 'react';
import ScrollArea from './ScrollArea';
import Languages from './Languages';
import InfoArea from './InfoArea';

class MainArea extends React.Component {

  render () {

    return (

      <main className="main__outer">
        <section className="main__intro">

            <InfoArea />

        </section>

        <ScrollArea
          showText={this.props.showText}
          bgs={this.props.bgs}
          blockContents={this.props.blockContents}
          language={this.props.language}
        />
        <Languages
          language={this.props.language}
          changeLanguage={this.props.changeLanguage}
        />
      </main>


    );
  }

}

export default MainArea;
