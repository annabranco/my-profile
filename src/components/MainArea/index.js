import React, { Component } from 'react';
import { Hero, ScrollArea, Languages } from '../';

export class MainArea extends Component {
  render() {
    return (
      <main className="main__outer">
        <div className="main__intro">
          <Hero
            texts={this.props.texts}
            language={this.props.language}
            viewedAll={this.props.viewedAll}
          />
        </div>

        <ScrollArea
          handleAnimation={this.props.handleAnimation}
          texts={this.props.texts}
          language={this.props.language}
          handleAdjustExpandedProjectsView={
            this.props.handleAdjustExpandedProjectsView
          }
          userViewedAllComponents={this.props.userViewedAllComponents}
        />

        {!this.props.doNotShowLanguagePopupAgain ? (
          <Languages
            language={this.props.language}
            texts={this.props.texts}
            changeLanguage={this.props.changeLanguage}
            clearLanguagePopup={this.props.clearLanguagePopup}
          />
        ) : null}
      </main>
    );
  }
}

export default MainArea;
