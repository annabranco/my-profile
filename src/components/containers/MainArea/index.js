import React, { useEffect } from 'react';
import { arrayOf, string, bool, func } from 'prop-types';
import { useStateWithLabel } from '../../../utils/hooks';
import ScrollArea from '../ScrollArea';
import { LanguagesModal } from '../../views';
import { LANG_MODAL_VISIBLE, BLOCK_LANG_MODAL } from '../../../constants';
import {
  experiencesTextPropType,
  formationPropType,
  languagesPropType,
  projectsPropType,
  skillGroupsPropType,
  textsPropType
} from '../../../types';

const MainArea = ({
  blockLangModal,
  closeLanguageModal,
  onChangeLanguage,
  displayThanks,
  experiences,
  formation,
  language,
  languages,
  projects,
  skills,
  texts,
  triggerThankYouMessage
}) => {
  const [hideForever, toggleHideForever] = useStateWithLabel(
    false,
    BLOCK_LANG_MODAL
  );
  const [langModalVisible, toggleModalVisible] = useStateWithLabel(
    false,
    LANG_MODAL_VISIBLE
  );

  useEffect(() => {
    if (!blockLangModal) {
      toggleModalVisible(true);
    }
  }, []);

  const toggleBlockLangModal = event =>
    toggleHideForever(event.currentTarget.checked);

  const onCloseLanguageModal = () => {
    toggleModalVisible(false);
    closeLanguageModal(hideForever);
  };

  return (
    <main>
      {langModalVisible && (
        <LanguagesModal
          hideForever={hideForever}
          languages={languages}
          languageSelected={language}
          onChangeLanguage={onChangeLanguage}
          onCloseLanguageModal={onCloseLanguageModal}
          texts={texts.languages}
          toggleBlockLangModal={toggleBlockLangModal}
        />
      )}
      <ScrollArea
        displayThanks={displayThanks}
        experiences={experiences}
        formation={formation}
        langModalVisible={langModalVisible}
        projects={projects}
        skills={skills}
        texts={texts}
        triggerThankYouMessage={triggerThankYouMessage}
      />
    </main>
  );
};

MainArea.propTypes = {
  blockLangModal: bool.isRequired,
  closeLanguageModal: func.isRequired,
  displayThanks: bool.isRequired,
  experiences: arrayOf(experiencesTextPropType).isRequired,
  formation: arrayOf(formationPropType).isRequired,
  language: string.isRequired,
  languages: arrayOf(languagesPropType).isRequired,
  onChangeLanguage: func.isRequired,
  projects: arrayOf(projectsPropType).isRequired,
  skills: arrayOf(skillGroupsPropType).isRequired,
  texts: textsPropType.isRequired,
  triggerThankYouMessage: func.isRequired
};

export default MainArea;
