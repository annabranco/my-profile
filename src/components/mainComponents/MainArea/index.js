import React from 'react';
import { bool, func } from 'prop-types';
import ScrollArea from '../ScrollArea';

const MainArea = ({
  isSeabedElementOpened,
  langModalVisible,
  openSeabedElement
}) => (
  <ScrollArea
    isSeabedElementOpened={isSeabedElementOpened}
    langModalVisible={langModalVisible}
    openSeabedElement={openSeabedElement}
  />
);

MainArea.propTypes = {
  langModalVisible: bool.isRequired,
  isSeabedElementOpened: bool.isRequired,
  openSeabedElement: func.isRequired
};

export default MainArea;
