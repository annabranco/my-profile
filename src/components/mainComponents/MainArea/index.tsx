import React, { Dispatch, ReactElement, SetStateAction } from 'react';
import { bool, func } from 'prop-types';
import ScrollArea from '../ScrollArea';

interface Props {
  isSeabedElementOpened: boolean;
  langModalVisible: boolean;
  openSeabedElement: Dispatch<SetStateAction<boolean>>;
}

const MainArea = ({
  isSeabedElementOpened,
  langModalVisible,
  openSeabedElement
}: Props): ReactElement => (
  <ScrollArea
    isSeabedElementOpened={isSeabedElementOpened}
    langModalVisible={langModalVisible}
    openSeabedElement={openSeabedElement}
  />
);

MainArea.propTypes = {
  langModalVisible: bool.isRequired,
  isSeabedElementOpened: bool.isRequired,
  openSeabedElement: func.isRequired
};

export default MainArea;
