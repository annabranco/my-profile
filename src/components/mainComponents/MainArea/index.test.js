import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import '../../../../testing/mocks/matchMedia';
import MainArea from '.';
import { setupMockProvider } from '../../../../testing/mocks';

// import { getMockState } from '../../../testing/mocks/mockState';

describe('< MainArea >', () => {
  let MockProvider;
  // let MOCK_STATE;
  let component;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    // MOCK_STATE = getMockState({ language: 'en' });

    component = shallow(
      <MockProvider>
        <MainArea langModalVisible={false} />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(component).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
