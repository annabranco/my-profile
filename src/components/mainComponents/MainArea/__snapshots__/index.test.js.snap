// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< MainArea > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MainArea
    langModalVisible={false}
  />
</Provider>
`;
